log/src/main/java/kcl/teamIndexZero/traffic/log/FileIO/FileOutput.java

```diff
@@ -15,66 +17,70 @@
+                MicroLogger.INSTANCE.log_Error("[FileOutput.FileOutput( ", folder_path, ", ", file_name, " )] Couldn't create/access the file.");
+                throw new IOException("[FileOutput.FileOutput( " + folder_path + ", " + file_name + " )] Failed to create/access the file " + getFilePath().toString());
             }
         } catch (IOException e) {
-            MicroLogger.INSTANCE.log_Error( "IOException raised in [FileOutput.FileOutput( ", folder_path, ", ", file_name, " )]");
-            MicroLogger.INSTANCE.log_ExceptionMsg( e );
+            MicroLogger.INSTANCE.log_Error("IOException raised in [FileOutput.FileOutput( ", folder_path, ", ", file_name, " )]");
+            MicroLogger.INSTANCE.log_ExceptionMsg(e);
             throw e;
-        } catch (InvalidPathException e ) {
-            MicroLogger.INSTANCE.log_Error( "InvalidPathException raised in [FileOutput.FileOutput( ", folder_path, ", ", file_name, " )]");
-            MicroLogger.INSTANCE.log_ExceptionMsg( e );
+        } catch (InvalidPathException e) {
+            MicroLogger.INSTANCE.log_Error("InvalidPathException raised in [FileOutput.FileOutput( ", folder_path, ", ", file_name, " )]");
+            MicroLogger.INSTANCE.log_ExceptionMsg(e);
             throw e;
         }
     }
 
     /**
      * Re-Opens the Writer to the same path/file
+     *
      * @return Success
      */
-    public boolean reOpenWriter() {
+    public synchronized boolean reOpenWriter() {
         try {
-            if( super.createFile() ) {
+            if (super.createFile()) {
                 this.writer = Files.newBufferedWriter(super.getFilePath(), super.getCharset(), StandardOpenOption.APPEND);
                 return true;
             } else {
                 return false;
             }
         } catch (IOException e) {
-            MicroLogger.INSTANCE.log_Error( "IOException raised in [FileOutput.reOpenWriter()] for , ", super.getFilePath());
-            MicroLogger.INSTANCE.log_ExceptionMsg( e );
+            MicroLogger.INSTANCE.log_Error("IOException raised in [FileOutput.reOpenWriter()] for , ", super.getFilePath());
+            MicroLogger.INSTANCE.log_ExceptionMsg(e);
             return false;
-        } catch (InvalidPathException e ) {
-            MicroLogger.INSTANCE.log_Error( "InvalidPathException raised in [FileOutput.reOpenWriter()] for , ", super.getFilePath() );
-            MicroLogger.INSTANCE.log_ExceptionMsg( e );
+        } catch (InvalidPathException e) {
+            MicroLogger.INSTANCE.log_Error("InvalidPathException raised in [FileOutput.reOpenWriter()] for , ", super.getFilePath());
+            MicroLogger.INSTANCE.log_ExceptionMsg(e);
             return false;
         }
     }
 
     /**
      * Closes the Writer
+     *
      * @return Success (
      */
-    public boolean closeWriter() {
-        if( this.writer != null ) {
+    public synchronized boolean closeWriter() {
```

Comment:
certainly not for our case, but I thought it may be useful for you for the future - as always with Java, there is a library for whatever you might think of. 

For the IO, there is nice set of IO utils from Apache  - having a nice [IOUtils.closeQuietly](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/IOUtils.html#closeQuietly(java.io.OutputStream)) method

Again - let's leave this as is, it's already very, very nice.